packages/serverpod_test/lib/src/test_serverpod.dart

```diff
@@ -94,7 +94,19 @@ class TestServerpod<T extends InternalTestEndpoints> {
   /// The test endpoints that are exposed to the user.
   T testEndpoints;
 
-  late final Serverpod _serverpod;
+  late final Serverpod Function() _buildServerpodAndInitializeEndpoints;
+
+  Serverpod? _serverpodInstance;
+  Serverpod get _serverpod {
+    var serverpodInstance = _serverpodInstance;
+    if (serverpodInstance != null) {
+      return serverpodInstance;
+    }
+
+    serverpodInstance = _buildServerpodAndInitializeEndpoints();
+    _serverpodInstance = serverpodInstance;
+    return serverpodInstance;
+  }
```

Comment:
Suggestion: I think you can replace this with this:

```suggestion
  Serverpod get _serverpod {
    return _serverpodInstance ??= _buildServerpodAndInitializeEndpoints();
  }
```

Reply:
Ahh, amazing! 🎉

Reply:
62d0959